File: problemset/Problem Set 6/sentimental-readability/readability.py
from cs50 import get_string

paragraph = get_string("Please entere a paragraph: ")


def countletters(paragraph):
    letters = 0
    for i in paragraph:
        if i.isalpha():
            letters = letters + 1
    return letters


def countwords(paragraph):
    words = 0
    for i in paragraph:
        if i == " ":
            words = words + 1
    return words + 1


def countsentences(paragraph):
    sentences = 0
    for i in paragraph:
        if i in (".", "?", "!"):
            sentences = sentences + 1
    return sentences


letters = countletters(paragraph)

words = countwords(paragraph)

sentences = countsentences(paragraph)

l = (float(letters) / float(words)) * 100
s = (float(sentences) / float(words) * 100)
index = 0.0588 * l - 0.296 * s - 15

if index < 1:
    print("Before Grade 1")
elif index >= 1 and index < 2:
    print("Grade 1")
elif index >= 2 and index < 3:
    print("Grade 2")
elif index >= 3 and index < 4:
    print("Grade 3")
elif index >= 4 and index < 5:
    print("Grade 4")
elif index >= 5 and index < 6:
    print("Grade 5")
elif index >= 6 and index < 7:
    print("Grade 6")
elif index >= 7 and index < 8:
    print("Grade 7")
elif index >= 8 and index < 9:
    print("Grade 8")
elif index >= 9 and index < 10:
    print("Grade 9")
elif index >= 10 and index < 11:
    print("Grade 10")
elif index >= 11 and index < 12:
    print("Grade 11")
elif index >= 12 and index < 13:
    print("Grade 12")
elif index >= 13 and index < 14:
    print("Grade 13")
elif index >= 14 and index < 15:
    print("Grade 14")
elif index >= 15 and index < 16:
    print("Grade 15")
elif index >= 16 and index < 17:
    print("Grade 16")
elif index > 17:
    print("Grade 16+")

File: problemset/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int countLetters(string txt);
int countWords(string txt);
int countSentences(string txt);

int main(void)
{
    // prompt the user for some text
    string text = get_string("Write a paragraph: ");

    // count the number of letters, words, and sentences in the text
    int letters = countLetters(text);
    // printf("Letters: %i\n", letters);

    int words = countWords(text);
    // printf("Words: %i\n", words);

    int sentences = countSentences(text);
    // printf("Sentences: %i\n", sentences);

    // compute the colemain-lieu index (to be implemented)
    float l = ((float) letters / words) * 100;
    float s = ((float) sentences / words) * 100;
    float index = 0.0588 * l - 0.296 * s - 15;
    // printf("index: %f\n", index);make
    
        // print the grade level (to be implemented)
    if (index < 1 )
    {
        printf("Below Grade 1\n");
    }
    else if (index >= 1 && index < 2)
    {
        printf("Grade 1\n");
    }
    else if (index >= 2 && index < 3)
    {
        printf("Grade 2\n");
    }
    else if (index >= 3 && index < 4)
    {
        printf("Grade 3\n");
    }
    else if (index >= 4 && index < 5)
    {
        printf("Grade 4\n");
    }
    else if (index >= 5 && index < 6)
    {
        printf("Grade 5\n");
    }
    else if (index >= 6 && index < 7)
    {
        printf("Grade 6\n");
    }
    else if (index >= 7 && index < 8)
    {
        printf("Grade 7\n");
    }
    else if (index >= 8 && index < 9)
    {
        printf("Grade 8\n");
    }
    else if (index >= 9 && index < 10)
    {
        printf("Grade 9\n");
    }
    else if (index >= 10 && index < 11)
    {
        printf("Grade 10\n");
    }
    else if (index >= 11 && index < 12)
    {
        printf("Grade 11\n");
    }
    else if (index >= 12 && index < 13)
    {
        printf("Grade 12\n");
    }
    else if (index >= 13 && index < 14)
    {
        printf("Grade 13\n");
    }
    else if (index >= 14 && index < 15)
    {
        printf("Grade 14\n");
    }
    else if (index >= 15 && index < 16)
    {
        printf("Grade 15\n");
    }
    else if (index >= 16 && index < 17)
    {
        printf("Grade 16\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
}

int countLetters(string txt)
{
    int letters = 0;
    for (int i = 0, length = strlen(txt); i < length; i++)
    {
        if (isalpha(txt[i]))
        {
            letters++;
        }
    }
    return letters;
}

int countWords(string txt)
{
    int words = 0;
    for (int i = 0, length = strlen(txt); i < length; i++)
    {
        if (txt[i] == ' ')
        {
            words++;
        }
    }
    return words + 1;
}

int countSentences(string txt)
{
    int sentences = 0;
    for (int i = 0, length = strlen(txt); i < length; i++)
    {
        if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}
